homeassistant/components/ozw/node.py

```diff
@@ -0,0 +1,190 @@
+            if new_value not in (option["Label"], option["Value"]):
+                continue
+            payload = int(option["Value"])
+            value.send_value(payload)
+            return OZWValidationResponse.process_success(payload)
+
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_SUPPORTED,
+            f"Invalid value {new_value} for parameter {parameter_index}",
+        )
+
+    # Int, Byte, Short are always passed as int, Decimal should be float
+    if value.type in (
+        ValueType.INT,
+        ValueType.BYTE,
+        ValueType.SHORT,
+        ValueType.DECIMAL,
+    ):
+        try:
+            if value.type == ValueType.DECIMAL:
+                new_value = float(new_value)
+            else:
+                new_value = int(new_value)
+        except ValueError:
+            return OZWValidationResponse.process_fail_on_type(value, new_value)
+        if (value.max and new_value > value.max) or (
+            value.min and new_value < value.min
+        ):
+            return OZWValidationResponse.process_fail(
+                ERR_NOT_SUPPORTED,
+                (
+                    f"Value {new_value} out of range for parameter "
+                    f"{parameter_index} (Min: {value.min} Max: {value.max})"
+                ),
+            )
+        value.send_value(new_value)
+        return OZWValidationResponse.process_success(new_value)
+
+    # This will catch BUTTON, STRING, and UNKNOWN ValueTypes
+    return OZWValidationResponse.process_fail(
+        ERR_NOT_SUPPORTED,
+        f"Value type of {value.type} for parameter {parameter_index} not supported",
+    )
+
+
+def get_config_parameters(node):
+    """Get config parameter from a node."""
+    command_class = node.get_command_class(CommandClass.CONFIGURATION)
+    if not command_class:
+        return OZWValidationResponse.process_fail(
+            ERR_NOT_FOUND, "Configuration parameters for OZW Node Instance not found"
+        )
+
+    values = []
+
+    for value in command_class.values():
+        value_to_return = {}
+        # BUTTON types aren't supported yet, and STRING and UNKNOWN
+        # are not valid config parameter types
+        if value.read_only or value.type in (
+            ValueType.BUTTON,
+            ValueType.STRING,
+            ValueType.UNKNOWN,
+        ):
+            continue
+
+        value_to_return = {
+            ATTR_LABEL: value.label,
+            ATTR_TYPE: value.type.value,
+            ATTR_CONFIG_PARAMETER: value.index.value,
+        }
+
+        if value.type == ValueType.BOOL:
+            value_to_return[ATTR_VALUE] = value.value
+
+        elif value.type == ValueType.LIST:
+            value_to_return[ATTR_VALUE] = value.value["Selected"]
+            value_to_return[ATTR_OPTIONS] = value.value["List"]
+
+        elif value.type in (
+            ValueType.INT,
+            ValueType.BYTE,
+            ValueType.SHORT,
+            ValueType.DECIMAL,
```

Comment:
Sorry, I think something got mixed up in the conversation. What I originally said is that Decimal is **not** a valid configuration parameter type. It is simply a defined value in py-ozw-mqtt.

The valid types are:
- BitSet
- Bool
- Byte
- Short
- Int
- List
- Button

Invalid types would therefore be:
- Decimal
- Schedule
- String
- Raw

You can see [here](https://github.com/OpenZWave/open-zwave/blob/189b9050ee857bc177a26973cd99b81ebc9fcd92/cpp/src/command_classes/Configuration.cpp#L167) which types are implemented.

py-ozw-mqtt does not define enums for:
- Schedule
- Raw
- BitSet

which would show up here as `UNKNOWN`. Of those, BitSet is the only config param type.

Reply:
Ah, thanks for the clarification and the code reference, I am obviously not very well versed in the spec 😄